### PR TITLE
Conditional return types for getFeatureById

### PR DESCRIPTION
**Pull request by ahocevar**
Fixes #15378

### CODE REVIEW

**Comment by ahocevar** [comment]
Thanks for any review.

File: src/ol/source/Vector.js [hunk 1]
```diff
@@ -275,7 +280,7 @@ class VectorSource extends Source {
     /**
      * A lookup of features by id (the return from feature.getId()).
      * @private
-     * @type {!Object<string, FeatureClass|Array<RenderFeature>>}
+     * @type {!Object<string, import('../Feature.js').FeatureLike|Array<import('../Feature.js').FeatureLike>>}
```
Comment (jahow) [inline src/ol/source/Vector.js:283]: I'm just curious why you didn't use `FeatureClassOrArrayOfRenderFeatures<FeatureClass>` here instead?

**Review by jahow — APPROVED** [review]
Thanks, interesting usage of conditional types!

File: src/ol/source/Vector.js (hunk 1 above)
Comment (ahocevar) [reply src/ol/source/Vector.js:283]: I had tried, but TypeScript didn't like it.

**Pull request merged**